freetype/__init__.py

```diff
@@ -977,17 +977,25 @@ def __init__( self, filename, index = 0 ):
         self._FT_Face = None
         #error = FT_New_Face( library, filename, 0, byref(face) )
         self._filebodys = []
-        try:
-            u_filename = c_char_p(_encode_filename(filename))
-            error = FT_New_Face( library, u_filename, index, byref(face) )
-        except UnicodeError:
-            with open(filename, mode='rb') as f:
-                filebody = f.read()
+        if hasattr(path_or_file, 'read'):
+            filebody = path_or_file.read()
             error = FT_New_Memory_Face( library, filebody, len(filebody),
                                         index, byref(face) )
             self._filebodys.append(filebody)  # prevent gc
+            self._filename = ":memory:"
+        else:
+            filename = path_or_file
+            try:
+                u_filename = c_char_p(_encode_filename(filename))
+                error = FT_New_Face( library, u_filename, index, byref(face) )
+            except UnicodeError:
```

Comment:
Hm. Would it make sense to just always forego FT_New_Face and use FT_New_Memory_Face instead and let Python handle the opening? One code path less.